sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -1237,8 +1639,10 @@ public Object restriction() {
               .setTransformId(pTransformId)
               .setInputId(mainInputId)
               .setElement(bytesOut.toByteString());
-      // We don't want to change the output watermarks or set the checkpoint resume time since
-      // that applies to the current window.
+      if (!outputWatermarkMapForUnprocessedWindows.isEmpty()) {
```

Comment:
We should leave the comment since it still makes sense. We could update it to just state that we are using the initial watermark for the output watermarks.